### DISCUSSION TITLE
Sqitch to install on GitHub Actions (no Docker image)

### DISCUSSION THREAD

**Discussion by Igor-Kuzenkov** [theory/sqitch#736] [Q&A]
Hello There!

I am trying to install Sqitch for Snowflake from the distribution package (e.g. https://github.com/sqitchers/sqitch/releases/download/v1.3.1/App-Sqitch-v1.3.1.tar.gz) by job on `ubuntu-latest` of GitHub Actions workflow.

Note, I do not want to use Sqitch Docker image, I want to use a standalone installation.

There is a guide in this repo how to install Sqitch (https://github.com/sqitchers/sqitch#installation), however, it does not works for me. It's definitely not enough to run only those commands. There are some prerequisites/dependences to be installed on GitHub Actions environment before the Sqitch installation (e.g. Build, cpan, etc.). When I try to set them up, I get a new portion of errors/exceptions. So far no luck.

So, my question is:
Is there a guide for installing Sqitch for Snowflake in a GitHub Actions environment (w/o Docker)?

**Comment by theory** [comment]
There is not, though it might be interesting. The project itself does this, though, using the Perl action and installing what it needs, as you can see in the [snowflake workflow](/sqitchers/sqitch/blob/develop/.github/workflows/snowflake.yml). If you were willing to use Docker images, it might be faster to just use the [Sqitch image](https://hub.docker.com/r/sqitch/sqitch) and then as part of the workflow add the Snowflake dependencies, as the [sqitch-snowflake Dockerfile](/sqitchers/docker-sqitch/blob/main/snowflake/Dockerfile) does.

**Reply by Igor-Kuzenkov** [reply]
@theory, Thank you for the suggestion!

Eventually I used `snowflake.sh` to install `odbc` and `snowsql` on GitHub Actions agent.
Then I used command
`apt-get -y install sqitch libdbd-odbc-perl snowflake-client`
to install sqitch with snowflake client.

It works for me and hope this is the correct way.

The only question I have is about the Sqitch version installed by `apt-get`. It's `v1.2.1`. Are you (or someone else) going to update the Sqitch package to the latest one (`1.3.1` today)?

**Reply by theory** [reply]
I do not maintain Debian's Sqitch package. I'm not sure who does, as [the search](https://search.debian.org/cgi-bin/omega?DB=en&P=sqitch) returns 0 results, and the [original link](https://packages.debian.org/stretch/sqitch) is now a 404.
